packages/bbui/src/Form/Core/DatePicker/TimePicker.svelte

```diff
@@ -1,46 +1,27 @@
 <script>
-  import { cleanInput } from "./utils"
-  import dayjs from "dayjs"
   import NumberInput from "./NumberInput.svelte"
   import { createEventDispatcher } from "svelte"
 
   export let value
 
   const dispatch = createEventDispatcher()
 
-  $: displayValue = value || dayjs()
+  $: displayValue = value?.format("HH:mm")
 
-  const handleHourChange = e => {
-    dispatch("change", displayValue.hour(parseInt(e.target.value)))
+  const handleChange = e => {
+    const [hour, minute] = e.target.value.split(":").map(x => parseInt(x))
+    dispatch("change", value.hour(hour).minute(minute))
```

Comment:
Value can be `null` here (throws an error if you try to change the time when no date is selected), so we'll need to just do something similar to `(value || dayjs())` like we used to have.

Reply:
This was not hit for some reason, but I added the check just in case. The existing behaviour treats default dates as empty and time as `now`. This seems slightly confusing, also, deleting the time value with the `backspace` key sets the value to 0, and the `clear` button clears the date but sets the time to `now` again.
The updated behaviour sets both the date and the time to undefined, so time will be `--:--` (HTML input standard). `Clear` button and `backspace` will remove any time value, getting `--:--` back.

Do you thing it's better to keep the existing behaviour? If so, I can just tweak it to behave like that

Reply:
Clearing it is more correct so that's good. The old behaviour didn't actually set the time to now when cleared, but just always displayed something (mostly to avoid this issue of trying to mutate a null time). I'm happy to keep the behaviour of actually clearing the field, as long as it's possible to edit the time from empty state without throwing an error. I'll give this another test now after your changes 👍

Reply:
This still throws an error when the value is null - we need to just update the change handler to use a new dayjs instance whenever we type into a time field that's empty.
![image](https://github.com/Budibase/budibase/assets/9075550/8a132af7-2038-4a78-b12c-1cd5ef0376ec)
![image](https://github.com/Budibase/budibase/assets/9075550/d55730c2-3ae3-4f12-9d4e-b8599f4bf33f)
Your original code here was totally fine, but just needed updated to `(value || dayjs()).hour...` like I mentioned.

Reply:
Good catch, thx!